src/client.ts

```diff
@@ -727,6 +727,81 @@ export class MongoClient<U extends BaseMongoObject, L extends BaseMongoObject> {
 		}
 	}
 
+	public async findOneByIdAndRemoveLockSubparts(
+		id: string,
+		baseLockFieldPath: string,
+		userId: string,
+	): Promise<U> {
+		try {
+			const query: FilterQuery<any> = {
+				_id: MongoUtils.oid(id),
+			};
+			return await this.findOneAndRemoveLockSubparts(query, baseLockFieldPath, userId);
+		} catch (e) {
+			LangUtils.throwN9ErrorFromError(e, { id, baseLockFieldPath, userId });
+		}
+	}
+
+	public async findOneByKeyAndRemoveLockSubparts(
+		keyValue: any,
+		baseLockFieldPath: string,
+		userId: string,
+		keyName: string = 'code',
+	): Promise<U> {
+		try {
+			const query: FilterQuery<any> = {
+				[keyName]: keyValue,
+			};
+			return await this.findOneAndRemoveLockSubparts(query, baseLockFieldPath, userId);
+		} catch (e) {
+			LangUtils.throwN9ErrorFromError(e, {
+				keyValue,
+				baseLockFieldPath,
+				userId,
+				keyName,
+			});
+		}
+	}
+
+	public async findOneAndRemoveLockSubparts(
+		query: FilterQuery<U>,
+		baseLockFieldPath: string,
+		userId: string,
+	): Promise<U> {
+		try {
+			if (!this.conf.lockFields) {
```

Comment:
I think you can mutualize this function with `findOneAndRemoveLock`, they have a lot in commons. You could extract the commons part, the only difference is the update query, and the error name thrown.

Reply:
done!